backend/internal/projects/resources.go

```diff
@@ -0,0 +1,73 @@
+package projects
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/psanford/memfs"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+// If srcURL is nil, get resources from the database.
+// If srcURL scheme is empty, get resources from the database. If none found, get from filesystem, if allowed.
+// If srcURL scheme is "file", get resources from filesystem, if allowed.
+// If srcURL scheme is "project", get resources from database.
+func (p Projects) openResourcesFS(ctx context.Context, projectID sdktypes.ProjectID, srcURL *url.URL) (fs.FS, error) {
+	if srcURL == nil || *srcURL == (url.URL{}) {
+		return p.openResourcesFS(ctx, projectID, &url.URL{Scheme: "project"})
+	}
+
+	switch srcURL.Scheme {
+	case "":
+		fs, err := p.openProjectResourcesFS(ctx, projectID, srcURL.Path)
+		if err != nil {
+			return nil, err
+		}
+		if fs != nil || !p.Config.Resources.AllowLocalFS {
```

Comment:
maybe split this condition to two? will make it more readable

```go
if fs != nil {
  return fs, nil
}

if !p.Config.Resources.AllowLocalFS {
  return nil, nil
}
```

Reply:
no longer relevant